core.js

```diff
@@ -5,83 +5,165 @@ var each = require("can-util/js/each/each");
 var assign = require("can-util/js/assign/assign");
 var isEmptyObject = require("can-util/js/is-empty-object/is-empty-object");
 var canLog = require("can-util/js/log/log");
+var canDev = require("can-util/js/dev/dev");
 require("./store");
 
 var fixtures = [];
 exports.fixtures = fixtures;
 
-// Adds a fixture to the list of fixtures.
-exports.add = function (settings, fixture) {
-	// When a fixture is passed a store like:
-	// `fixture("/things/{id}", store)`
-	if(fixture && (fixture.getData || fixture.getListData)) {
-		var root = settings,
-			store = fixture,
-			idProp = store.idProp,
-			itemRegex = new RegExp('\\/\\{' + idProp+"\\}.*" ),
-			rootIsItemUrl = itemRegex.test(root),
-			getListUrl = rootIsItemUrl ? root.replace(itemRegex, "") : root,
-			getItemUrl = rootIsItemUrl ? root : (root.trim() + "/{" + idProp + "}");
-		fixture = undefined;
-		settings = {};
-		settings["GET "+getItemUrl] = store.getData;
-		settings["DELETE "+getItemUrl] = store.destroyData;
-		settings["PUT "+getItemUrl] = store.updateData;
-		settings["GET "+getListUrl] = store.getListData;
-		settings["POST "+getListUrl] = store.createData;
-	}
+function isStoreLike (fixture) {
+	return fixture && (fixture.getData || fixture.getListData);
+}
 
-	// If fixture is provided, set up a new fixture.
-	if (fixture !== undefined) {
+var methodMapping = {
+	item: {
+		'GET': 'getData',
+		'PUT': 'updateData',
+		'DELETE': 'destroyData',
+	},
+	list: {
+		'GET': 'getListData',
+		'POST': 'createData'
+	}
+};
 
-		if (typeof settings === 'string') {
+function getMethodAndPath (route) {
+	// Match URL if it has GET, POST, PUT, DELETE or PATCH.
+	var matches = route.match(/(GET|POST|PUT|DELETE|PATCH) (.+)/i);
```

Comment:
It looks to be [using tabs](https://raw.githubusercontent.com/canjs/can-fixture/ac8757774871020024b34f34b5dd41884297bdd6/core.js).